internal/tofu/node_resource_abstract.go

```diff
@@ -290,33 +301,88 @@ func (n *NodeAbstractResource) DependsOn() []*addrs.Reference {
 	return result
 }
 
-func (n *NodeAbstractResource) SetProvider(p addrs.AbsProviderConfig) {
-	n.ResolvedProvider = p
+func (n *NodeAbstractResource) SetPotentialProviders(potentialProviders []distinguishableProvider) {
+	n.potentialProviders = potentialProviders
```

Comment:
do we also want to check if ResolvedResourceProvider is nil? or maybe set it to nil?